config.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { schema, TypeOf } from '@osd/config-schema';
+
+export const configSchema = schema.object({
+  enabled: schema.boolean({ defaultValue: true }),
+  chat: schema.object({
+    enabled: schema.boolean({ defaultValue: true }),
+  }),
```

Comment:
Do we need two `enabled` flag?

Reply:
`plugin_id.enabled` is actually default behavior to "install" itself when the plugin discovery process is occurring. the current `chat.enabled` here will still make all this code still be available in the browser but it doesn't do the setup steps.

however, by adding this enabled. the plugin will not be discovered and hosted in the browser. if the chat is not enabled and the expected behavior is none of the code gets added to the browser then we should opt for removing the `chat.enabled`